draft-ietf-wimse-workload-identity-practices.md

```diff
@@ -158,6 +158,44 @@ The figure outlines the following steps which are applicable in any pattern.
 
+## Credential issuance
+
+Credentials can be provisioned to the workload through different mechanisms which present different complexity and security risks. The following section highlights the pros and cons of most widespread solutions.
+
+### Enviromental variable
+
+Injecting the credentials into the enviromental variables allows for simple and fast deployments. Applications can directly access them through system-level mechanism, e.g., through the env command in linux. This flexibility, however, comes wiht security drawbacks.
+
+* 1) enviromental variables are static in nature and more dynamic solutions require to introduces higher complexitiy.
+  
+* 2) performing access control to enviromental variable is not trivial and it might also not reach the same security results.
+  
+* 3) enviromental information are often used for debugging purpose and can be printed at start-up or inside logs.
+
+Leveraging enviromental variable to provide credentials should be favored when the provided secrets have a short-term validity, i.e., an initial secret during the set-up of the application, and/or when simplicity is required, e.g., during the development of PoC.
+
+### Volumes
+
+Volumes, e.g., projected volumes in Kubernetes, allow to inject credential to the container through the file-system. This solution offers higher degree of flexibility thanks to the possibility of dynamically rotate the secrets and perform access control on the injected file. This additional security comes with additional complexity.
+
+* 1) access control to the mounted volume needs to be correctly configure to limit access from unwanted applications. Solution such as DAC (uid and guid) or MAC (seLinux) are avaible in most Operating Systems.
+
+* 2) isolating the mounted volumes ,e.g., through namespaces, is required to avoid that a compromised application is able to escape to the main OS and ,for example, retrieve private information of other processes.
+
+* 3) credentials rotation requires a monitoring solution to detect near-to-expiration secrets and subtitute them. This is usually embedded into orchestrators such as Kubernetes, but requires proper configuration.
+
+Leveraging volumes to provision secrets offers all the required security tools, at the cost of a more complex configuration. This approach should be favored when multiple process might require to access the same set of credentials.
+
+### Network inferaces
+
+Network interfaces rely on the network stack to communicate between the Host and the container application. Some examples of this interface are UNIX Domain Socket (spiffe), loopback interface, Magic (Link-Local) Address (AWS Metadata service). These solutions offer on-demand identity rotation and ensure security thanks to network isolation, but introduce new complexity.
```

Comment:
> These solutions offer on-demand identity rotation.

I see multiple flavors at the `on-demand` part. Either these APIs "push" new credentials actively to the workload (require some notification mechanism or open channel) or the workload itself needs to have logic to "re-request".

Some ideas from the top of mind:
> Local APIs offer the capability to re-provision updated credentials. Communication between workload and API allows the workload to re-request a new credential (or a different one). Based on the technology it is even possible to pro-actively distribute new credentials to workloads (e.g. upon expiry, during a revocation event or a change in permissions.